pkg/labels/selector_builder.go

```diff
@@ -0,0 +1,38 @@
+package labels
+
+import (
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+)
+
+type selectorBuilder struct {
+	selector labels.Selector
+}
+
+func SelectorBuilder() selectorBuilder {
+	return selectorBuilder{
+		selector: labels.NewSelector(),
+	}
+}
+
+func (o selectorBuilder) WithComponent(name string) selectorBuilder {
+	req, err := labels.NewRequirement(componentLabel, selection.Equals, []string{name})
+	if err != nil {
+		panic(err)
```

Comment:
Is it okay to panic ?

Reply:
I think it is. The only error that can happen is when the number of values in the last param's array does not match with the operation in the second argument. Here it will never fail because we need one value with the `Eq` operation, and we want to fail fast if a developer changes the operation or the last param in a wrong way.